classes/spec/book_spec.rb

```diff
@@ -0,0 +1,37 @@
+require_relative 'spec_helper'
+
+describe Book do
+  before :each do
+    @book = Book.new('Afghanistan', 'Abdulali')
+    @teacher = Teacher.new('math', 30, 'Salim')
+    @rental = Rental.new('2023-08-02', @book, @teacher, [0, 0])
+  end
+
+  context 'Add new book:' do
+    it 'Check if the @book is an object of the Book' do
+      expect(@book).to be_an_instance_of Book
+    end
+    it 'Check for title' do
+      expect(@book.title).to eq 'Afghanistan'
+    end
+    it 'Check for author' do
+      expect(@book.author).to eq 'Abdulali'
+    end
+  end
+
+  context 'Change the data to HASH format:' do
+    it 'Check if the book object is converted to JSON format' do
+      hash_obj = @book.to_hash.to_s
+      expect(hash_obj).to eq '{"title"=>"Afghanistan", "author"=>"Abdulali", "rentals"=>[{"date"=>"2023-08-02"}]}'
+    end
+    it 'Check if the array is converted into hash' do
+      expect(@book.to_hash.class).to eq Hash
+    end
+  end
+
+  context 'When adding a new rentals' do
+    it 'The add_rental method returns a rental and adds it to the book' do
+      expect(@book.rentals[0]).to eql @rental
+    end
+  end
```

Comment:
- [ ] I really can not see how you are testing the method `add_rental` kindly make this test to test the method. it should test it like so 
```
....
  rental = person.add_rental(book, "2021-05-04")

      expect(person.rentals).to eql [rental]
```